packages/contracts-bedrock/src/L2/SuperchainWETH.sol

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { WETH98 } from "src/dispute/weth/WETH98.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { L1Block } from "src/L2/L1Block.sol";
+import { IL2ToL2CrossDomainMessenger } from "src/L2/IL2ToL2CrossDomainMessenger.sol";
+import { ETHLiquidity } from "src/L2/ETHLiquidity.sol";
+import { ISuperchainERC20Extensions } from "src/L2/ISuperchainERC20.sol";
+import { ISemver } from "src/universal/ISemver.sol";
+
+import "src/libraries/errors/CommonErrors.sol";
+
+/// @title SuperchainWETH
+/// @notice SuperchainWETH is a version of WETH that can be freely transfered between chains within
+///         the superchain. SuperchainWETH can be converted into native ETH on chains that do not
+//          use a custom gas token.
+contract SuperchainWETH is WETH98, ISuperchainERC20Extensions, ISemver {
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0
+    string public constant version = "1.0.0-beta.1";
+
+    /// @inheritdoc WETH98
+    function deposit() public payable override {
+        if (L1Block(Predeploys.L1_BLOCK_ATTRIBUTES).isCustomGasToken()) revert NotCustomGasToken();
+        super.deposit();
+    }
+
+    /// @inheritdoc WETH98
+    function withdraw(uint256 wad) public override {
+        if (L1Block(Predeploys.L1_BLOCK_ATTRIBUTES).isCustomGasToken()) revert NotCustomGasToken();
+        super.withdraw(wad);
+    }
+
+    /// @inheritdoc ISuperchainERC20Extensions
+    function sendERC20(address dst, uint256 wad, uint256 chainId) public {
+        // Burn from user's balance.
+        _burn(msg.sender, wad);
+
+        // Burn to ETHLiquidity contract.
+        if (!L1Block(Predeploys.L1_BLOCK_ATTRIBUTES).isCustomGasToken()) {
+            ETHLiquidity(Predeploys.ETH_LIQUIDITY).burn{ value: wad }();
+        }
+
+        // Send message to other chain.
+        IL2ToL2CrossDomainMessenger(Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER).sendMessage({
+            _destination: chainId,
+            _target: address(this),
+            _message: abi.encodeCall(this.relayERC20, (dst, wad))
+        });
+
+        // Emit event.
+        emit SendERC20(msg.sender, dst, wad, chainId);
+    }
+
+    /// @inheritdoc ISuperchainERC20Extensions
+    function relayERC20(address dst, uint256 wad) external {
+        // Receive message from other chain.
+        IL2ToL2CrossDomainMessenger messenger = IL2ToL2CrossDomainMessenger(Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER);
```

Comment:
You could make `messenger` an internal constant instead of defining it inline here

Reply:
Solidity won't let you make it an internal constant because `Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER` is an imported value which messes with the compiler.